chore(engine): Add Support for skipIoMappings

psavidis opened the pull request:
Parameter is added to

- deleteProcessInstances, deleteProcessInstancesAsync
- The respective DeleteBatch commands that the methods issue and their config

danielkelemen commented on engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/ProcessInstanceRestServiceInteractionTest.java:
```diff
@@ -653,7 +678,59 @@ public void testDeleteAsyncHistoricQueryBasedWithSkipSubprocesses() {
+  @Test
+  public void testDeleteAsyncHistoricQueryBasedWithSkipIoMappingsTrue() {
+    when(runtimeServiceMock.deleteProcessInstancesAsync(
+        eq((List<String>)null),
```
❓Are these casts necessary?

psavidis replied:
They weren't, removed in last commit.

danielkelemen commented on engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/ProcessInstanceRestServiceInteractionTest.java:
```diff
@@ -653,7 +678,59 @@ public void testDeleteAsyncHistoricQueryBasedWithSkipSubprocesses() {
+        anyBoolean(),
+        anyBoolean(),
+        eq(true)
+    ))
+        .thenReturn(new BatchEntity());
```
🙃 Looks a bit weird.

```suggestion
    )).thenReturn(new BatchEntity());
```

psavidis replied:
Applied suggestion.

danielkelemen reviewed (commented):
Looks pretty good! 👍
I adde some formatting hints and a question about a possible simplification.

danielkelemen commented:
❓Should we add the skip flag in the [`deleteAsync`](https://github.com/camunda/camunda-bpm-platform/blob/3934-feature-add-param-to-engine/engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/impl/ProcessInstanceRestServiceImpl.java#L147) too?

psavidis commented on engine/src/main/java/org/camunda/bpm/engine/impl/cmd/AbstractDeleteProcessInstanceCmd.java:
```diff
@@ -54,6 +54,7 @@ public abstract class AbstractDeleteProcessInstanceCmd {
   protected boolean skipCustomListeners;
   protected boolean skipSubprocesses;
   protected boolean failIfNotExists = true;
+  protected boolean skipIoMappings;
```
I've removed field, it doesn't need to belong to the children class.

psavidis commented on engine/src/main/java/org/camunda/bpm/engine/impl/cmd/DeleteProcessInstancesCmd.java:
```diff
@@ -31,20 +30,31 @@ public class DeleteProcessInstancesCmd extends AbstractDeleteProcessInstanceCmd
+          externallyTerminated,
+          skipIoMappings,
+          skipSubprocesses
+      );
```
All the instance field properties are removed from the `deleteProcessInstance`.

psavidis commented on engine/src/test/java/org/camunda/bpm/engine/test/api/runtime/RuntimeServiceAsyncOperationsTest.java:
```diff
@@ -392,6 +391,54 @@ public void testDeleteProcessInstancesAsyncWithSkipSubprocesses() {
+    testRule.assertProcessEnded(instanceId1);
+    testRule.assertProcessEnded(instanceId2);
+
+    assertEquals(1, historyService.createHistoricVariableInstanceQuery().processInstanceId(instanceId1).list().size());
```
Considered to use it initially but couldn't find how AssertJ adds to the readability of this case. Still, i find the argument plausible for consistency sake and following the guide so i decided to use your suggestion.

psavidis commented:
> ❓Should we add the skip flag in the [`deleteAsync`](https://github.com/camunda/camunda-bpm-platform/blob/3934-feature-add-param-to-engine/engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/impl/ProcessInstanceRestServiceImpl.java#L147) too?
> 
> Update: the flag is already there as it uses the same `DTO`, it's just not passed to the service. I would also add tests for this API in the `ProcessInstanceRestServiceInteractionTest`.

Added the flag to `deleteAsync` as well. Corrected the failing tests due to using mocks (the flag is propagated now and the mocks were lacking it so i had to adjust the statements).

Also, i've added tests that include the `skipIoMappings` parameter that test the propagation using `true` and `false`.

danielkelemen reviewed (approved):
Looks good! 👍
